src/recipes/editor-plugin/inspector-plugins.md

```diff
@@ -0,0 +1,342 @@
+representation; if not, it's necessary to return `false`.
+This allows for specific control over where and how processing occurs.
+
+
+## Add Editor Plugin
+
+Only one thing left to do: define the editor plugin that will kick off all this magic!
+This can be a generic `EditorPlugin` or a more specific `InspectorEditorPlugin`, depending
+on what you want to achieve.
+
+
+```rust
+#[derive(GodotClass)]
+#[class(tool, init, editor_plugin, base=EditorPlugin)]
+struct RustEditorPlugin {
+    base: Base<EditorPlugin>,
+    random_inspector: Gd<RandomInspectorPlugin>,
+}
+```
+
+```rust
+#[godot_api]
+impl IEditorPlugin for RustEditorPlugin {
+    fn enter_tree(&mut self) {
+        // create our inspector plugin and save it to next remove
+        self.random_inspector = RandomInspectorPlugin::new_gd();
+        self.to_gd()
+            .add_inspector_plugin(self.random_inspector.to_variant().to());
+    }
+
+    fn exit_tree(&mut self) {
+        self.to_gd()
+            .remove_inspector_plugin(self.random_inspector.to_variant().to());
+    }
+}
+```
+
+
+```admonish danger title="Troubleshooting"
+Sometimes after compilation, you may encounter errors or panic. Most likely, all you need to do is simply **restart** the Godot Editor.
+```
+
+Example error:
+
+```log
+Initialize godot-rust (API v4.2.stable.official, runtime v4.2.2.stable.official)
+ERROR: Cannot get class 'RandomInspectorPlugin'.
+   at: (core/object/class_db.cpp:392)
+ERROR: Cannot get class 'RandomInspectorPlugin'.
+   at: (core/object/class_db.cpp:392)
+```
+
+
+# Full code
+
+
+```rust
+// file: addon.rs
+
+use godot::{
+    engine::{
+        Button, EditorInspectorPlugin, EditorPlugin, EditorProperty, IEditorInspectorPlugin,
+        IEditorPlugin, IEditorProperty,
+    },
+    global,
+    prelude::*,
+};
+use rand::Rng;
+
+#[derive(GodotClass)]
+#[class(tool, init, editor_plugin, base=EditorPlugin)]
+struct RustEditorPlugin {
+    base: Base<EditorPlugin>,
+    random_inspector: Gd<RandomInspectorPlugin>,
+}
+
+#[godot_api]
+impl IEditorPlugin for RustEditorPlugin {
+    fn enter_tree(&mut self) {
+        self.random_inspector = RandomInspectorPlugin::new_gd();
+        self.to_gd()
+            .add_inspector_plugin(self.random_inspector.to_variant().to());
+    }
+
+    fn exit_tree(&mut self) {
+        self.to_gd()
+            .remove_inspector_plugin(self.random_inspector.to_variant().to());
+    }
+}
+
+#[derive(GodotClass)]
+#[class(tool, init, base=EditorInspectorPlugin)]
+struct RandomInspectorPlugin {
+    base: Base<EditorInspectorPlugin>,
+}
+
+#[godot_api]
+impl IEditorInspectorPlugin for RandomInspectorPlugin {
+    fn parse_property(
+        &mut self,
+        _object: Gd<Object>,
+        value_type: VariantType,
+        name: GString,
+        _hint_type: global::PropertyHint,
+        _hit_name: GString,
+        _flags: global::PropertyUsageFlags,
+        _wide: bool,
+    ) -> bool {
```

Comment:
Let's stick to the official parameter names (for `value_type` it's OK since `type` is a keyword):
```suggestion
    fn parse_property(
        &mut self,
        _object: Gd<Object>,
        value_type: VariantType,
        name: GString,
        _hint_type: global::PropertyHint,
        _hint_string: GString,
        _usage_flags: global::PropertyUsageFlags,
        _wide: bool,
    ) -> bool {
```

Reply:
ok, but what is "hint string" - i don't really undestand this suffix. looks redundant